src/server/game/map/tile/errand.ts

```diff
@@ -45,7 +45,7 @@ export class WorkErrand {
     },
     [ErrandType.UNIT_TRAINING]: (world, map, tile, action) => {
       if (!(tile.owner && action.location)) return;
-      const newUnit = new Unit(action.option, tile.owner.civID, action.location);
+      const newUnit = new Unit(action.option, action.location, tile.owner.civID, tile.owner.civID ? undefined : tile.owner.id);
```

Comment:
So, is my assumption that what this is doing is ensuring that a unit can only have a civID or_ a (city)id correct?

Reply:
Correct.